src/reflect/scala/reflect/internal/SymbolPairs.scala

```diff
@@ -35,7 +35,7 @@ abstract class SymbolPairs {
    *  when viewed from IterableClass.
    */
   def sameInBaseClass(baseClass: Symbol)(tp1: Type, tp2: Type) =
-    (tp1 baseType baseClass) =:= (tp2 baseType baseClass)
+    tp1.baseType(baseClass).typeSymbol == tp2.baseType(baseClass).typeSymbol
```

Comment:
We're basically computing type equality under erasure, right? I guess typeSymbol is a good approximation, but this would still break where the erasure is different (Arrays?)

Reply:
I think looking at typeSymbol could be the correct thing here, `hasCommonParentAsSubclass` checks ``Do `sym1` and `sym2` have a common subclass in `parents` ``, so it talks about classes, not types.